common/xfer/pipes.go

```diff
@@ -27,7 +27,16 @@ type pipe struct {
 	onClose         func()
 }
 
-// NewPipe makes a new... pipe.
+// NewPipeFromEnds makes a new pipe specifying its ends
+func NewPipeFromEnds(local io.ReadWriter, remote io.ReadWriter) Pipe {
```

Comment:
(what I'm getting at: this type is already used in too many places for too many things, and needs to be broken down into multiple types for different uses)